File: java-playground/src/test/java/org/gonza/javaplayground/car/CarTest.java
package org.gonza.javaplayground.car;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CarTest {
    @Test
    void Car는_name을_가질_수_있다() {
        String name = "NAME";

        Car car = new Car(name);

        assertEquals(car.getName(), name);
    }

    @Test
    void Car는_name이_5자를_초과할_수_없다() {
        String oneLengthName = "1";
        String fiveLengthName = "12345";
        String sixLengthName = "123456";

        assertDoesNotThrow(() -> new Car(oneLengthName));
        assertDoesNotThrow(() -> new Car(fiveLengthName));
        assertThrows(IllegalArgumentException.class, () -> new Car(sixLengthName));
    }

    @Test
    void Car가_생성되면_position은_0이다() {
        Car car = new Car("name");

        assertEquals(car.getPosition(), 0);
    }

    @Test
    void Car는_한_칸_이동할_수_있다() {
        Car car = new Car("name");
        assertEquals(car.getPosition(), 0);

        car.moveForward();

        assertEquals(car.getPosition(), 1);
    }
}

File: kotlin-playground/src/test/kotlin/org/gonza/kotlinplayground/domain/CarTest.kt
package org.gonza.kotlinplayground.domain

import Car
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertDoesNotThrow
import org.junit.jupiter.api.assertThrows

class CarTest {
    val name = "name"
    @Test
    fun `차는 이름을 받아 생성된다`() {
        val result = Car(name)
        assertThat(result.name).isEqualTo(name)
    }

    @Test
    fun `이름이 5자를 초과할 수 없다` () {
        val name = "longerthanfive"
        assertThrows<IllegalArgumentException> { Car(name) }
    }

    @Test
    fun `차는 최초로 생성될 시 position이 0이다` () {
        val result = Car(name)
        assertThat(result.position).isEqualTo(0)
    }

    @Test
    fun `차는 한 칸씩 움직인다`() {
        val result = Car(name)
        assertThat(result.position).isEqualTo(0)
        result.move()
        assertThat(result.position).isEqualTo(1)
    }
}